src/module.c

```diff
@@ -1135,7 +1169,105 @@ int moduleArgFlagsConvert(int flags) {
     return realflags;
 }
 
-RedisModuleCommandArg *RM_CreateCommandArg(const char* argname, RedisModuleCommandArgType type, const char *token, const char *summary, const char* since, int flags, const char *value) {
+/* Create an opaque structure representing a command argument
+ * (For manipulating the output of COMMAND)
+ *
+ * * `argname`: Name of the argument
+ * * `type`: The type of the argument.
+ *           Some types allow an argument to have sub-arguments
+ *           (REDISMODULE_ARG_TYPE_ONEOF and REDISMODULE_ARG_TYPE_BLOCK)
+ * * `key_spec_index` : If the `type` is REDISMODULE_ARG_TYPE_KEY you
+ *                      must provide the index of the key-spec associated
+ *                      with this argument. See RedisModule_AddCommandKeySpec.
+ *                      If the argument is not a key, you must specify -1.
+ * * `token`: The token preceding the argument (optional).
+ *            Example: the argument "seconds" in SET has a token "EX")
+ *            If the argument consists of only a token (Example: "NX" in SET)
+ *            the type should be REDISMODULE_ARG_TYPE_PURE_TOKEN and the
+ *            value should be NULL
+ * * `summary`: A short description of the argument (optional)
+ * * `since`: The first version which included this argument (optional)
+ * * `flags`: See REDISMODULE_CMD_ARG_*
+ * * `value`: The display-value of the argument. This string is what
+ *            should be displayed when creating the command syntax from
+ *            the output of COMMAND.
+ *            If `token` is not NULL, it should also be displayed.
```

Comment:
not specifically related to this block:
please run the script that generate an md from this doc and open it in some md viewer to see that the syntax is displayed correctly.

Reply:
how do i do that?

Reply:
`ruby modules/gendoc.rb` to generate the md file.

you can maybe post it in GH gist editor or something, but the truth is that redis-io renders md files a bit differently, so try to follow this:
```
# prerequisits
sudo apt-get install ruby-rack ruby-dev
sudo gem install dep
sudo dep install
run a plain redis-server (no password, default port)
# then
inside redis-io repo you run:
REDIS_DOC=/home/oran/work/redis-doc rackup 
```

you can probably ask Yoav / Itamar / Viktor for help